### PR TITLE
Fix memory leaks

### LINKED ISSUE #1048: Memory leak when going from compress screen to home screen
I don't know this is whether or not a bug
once start compress image process will greatly improve memory usage

any solution can reduce memoery usage or not?

as you can see I just opened the [website](https://squoosh.app/)
the tab take about 30mb of memory 
image
When I upload a image about 1.8m, the memory is straight up, and then wait for compress done (using default options)
the memory is about 800mb
and then i close the website image editor, the memory is about 350mb usage, and cant no longer reduce
image
this is right or not? and there is a way to solve this or not? thank you guys

### PR DESCRIPTION
**Pull request by jakearchibald**
Fixes #1048

### CODE REVIEW

File: src/client/lazy-app/Compress/Output/custom-els/TwoUp/index.ts [hunk 1]
```diff
@@ -84,20 +77,36 @@ export default class TwoUp extends HTMLElement {
...
+
     if (!this._everConnected) {
       this._resetPosition();
       this._everConnected = true;
     }
   }
 
+  disconnectedCallback() {
+    window.removeEventListener('keydown', this._onKeyDown);
+    window.removeEventListener('resize', this._onResize);
```
Comment (jakearchibald) [inline src/client/lazy-app/Compress/Output/custom-els/TwoUp/index.ts:98]: We should have caught these at review time 😄. I guess we would have if it was a Preact component, but we're less experienced with web components.

File: src/client/lazy-app/Compress/Output/custom-els/TwoUp/index.ts [hunk 2]
```diff
@@ -84,20 +77,36 @@ export default class TwoUp extends HTMLElement {
...
+    if (this._resizeObserver) this._resizeObserver.disconnect();
```
Comment (jakearchibald) [inline src/client/lazy-app/Compress/Output/custom-els/TwoUp/index.ts:99]: I really don't think we should have to do this, but it's a direct cause of the leak. Filed https://bugs.chromium.org/p/chromium/issues/detail?id=1220041.

File: src/client/lazy-app/Compress/index.tsx [hunk 3]
```diff
@@ -377,6 +377,7 @@ export default class Compress extends Component<Props, State> {
 
   componentWillUnmount(): void {
     updateDocumentTitle({ loading: false });
+    this.widthQuery.removeListener(this.onMobileWidthChange);
```
Comment (jakearchibald) [inline src/client/lazy-app/Compress/index.tsx:380]: This doesn't seem to be causing a leak. Or, if it is, it's small. But, good to add it for safety.

**Review by surma — APPROVED** [review]
Nice work. LGTM.

File: src/client/lazy-app/Compress/index.tsx (hunk 3 above)
Comment (jakearchibald) [reply src/client/lazy-app/Compress/index.tsx:380]: (I was wrong here, it was causing a leak, it was just hard to see due to the resizeObserver issue)

**Pull request merged**